python_modules/libraries/dagster-mlflow/dagster_mlflow/resources.py

```diff
@@ -218,6 +227,30 @@ def chunks(params: dict, size: int = 100):
             yield {k: params[k] for k in islice(it, size)}
 
 
+class MLFlowTrackingResource(ConfigurableResource):
+    experiment_name: str = PyField(description="MlFlow experiment name.")
+    mlflow_tracking_url: Optional[str] = PyField(
+        default=None, description="MlFlow trakcing server URI."
+    )
+    parent_run_id: Optional[str] = PyField(
+        default=None, description="MlFlow run ID of parent run, if this is a nested run."
+    )
+    env: Dict[str, str] = PyField(default={}, description="Environment variables for MlFlow setup.")
+    env_to_tag: Optional[List[str]] = PyField(
+        default=None, description="List of environment variables to log as tags in MlFlow."
+    )
+    extra_tags: Dict[str, str] = PyField(
+        default={}, description="Any extra key-value tags to log in MlFlow."
+    )
+
+    @contextmanager
+    def get_tracking_client(self) -> Generator[MlFlowTrackingClient, None, None]:
+        context = self.get_resource_context()
+        client = MlFlowTrackingClient(context)
+        yield client
+        client.cleanup_on_error()
```

Comment:
`cleanup_on_error` should be in a finally block.

Also in terms of naming convention, what would you think about `open_tracking_client` to indicate that it is a context manager?